core/src/com/punchbrain/awake/behavior/HoundBehavior.java

```diff
@@ -27,15 +30,34 @@
     private final Queue<Vector2> snapshots = new LinkedList<Vector2>();
     private final GameEventManager eventManager;
     private final LightingManager lightingManager;
+    private final CircuitModelMap circuitModelMap;
+    private boolean inLight;
 
-    public HoundBehavior(GameObject player, GameEventManager eventManager, LightingManager lightingManager) {
+    public HoundBehavior(GameObject player, GameEventManager eventManager, LightingManager lightingManager, CircuitModelMap circuitModelMap) {
         this.player = player;
         this.eventManager = eventManager;
         this.lightingManager = lightingManager;
+        this.circuitModelMap = circuitModelMap;
     }
 
     @Override
     public void update(final GameObject hound, final float delta) {
+
+        // TODO: inject a lookup map to find circuit by game object
+        // TODO: introduce object to model stereotype Map<GameObject, Model> which can be added to the context
+        // Map<Model.class,Map<GameObject, Model>>
```

Comment:
very interesting idea. We might wanna add this to braingdx if it fulfills these criterias:

- it saves a lot of time when creating new games
- it does not introduce strange coupling at scale (checkout https://github.com/bitbrain/galacticum as a reference, this is a project of mine where I attempted to create an infinite 2D space game but it failed due to coupling issues. I want to avoid them in braingdx)

Let's see some code in action and  see if it's useful!